contracts/utils/Strings.sol

```diff
@@ -391,11 +391,11 @@ library Strings {
         uint256 end
     ) internal pure returns (bool success, address value) {
         // check that input is the correct length
-        bool hasPrefix = (begin < end + 1) && bytes2(_unsafeReadBytesOffset(bytes(input), begin)) == bytes2("0x"); // don't do out-of-bound (possibly unsafe) read if sub-string is empty
+        bool hasPrefix = (end > begin + 1) && bytes2(_unsafeReadBytesOffset(bytes(input), begin)) == bytes2("0x"); // don't do out-of-bound (possibly unsafe) read if sub-string is empty
 
         uint256 expectedLength = 40 + hasPrefix.toUint() * 2;
 
-        if (end - begin == expectedLength && end <= bytes(input).length) {
+        if (begin <= end && end - begin == expectedLength && end <= bytes(input).length) {
```

Comment:
Since the issue here is the overflow, it feels like an equivalent but cheper option would be
```suggestion
        if (begin + expectedLength == end && end <= bytes(input).length) {
```